addon/components/stacked-vertical-bar-chart.js

```diff
@@ -0,0 +1,1015 @@
+      .domain(this.get('barNames'))
+      .rangeRoundBands([0, this.get('graphicWidth')],
+                       betweenBarPadding / 2,
+                       betweenBarPadding / 2);
+  }),
+
+  // Override axis mix-in min and max values to listen to the scale's domain
+  minAxisValue: Ember.computed('yScale', function() {
+    var yScale = this.get('yScale');
+    return yScale.domain()[0];
+  }),
+
+  maxAxisValue: Ember.computed('yScale', function() {
+    var yScale = this.get('yScale');
+    return yScale.domain()[1];
+  }),
+
+  // ---------------------------------------------------------------------------
+  // Color Configuration
+  // ---------------------------------------------------------------------------
+
+  numColorSeries: Ember.computed.alias('allSliceLabels.length'),
+
+  sliceColors: Ember.computed('allSliceLabels.[]', 'getSeriesColor', function() {
+    var fnGetSeriesColor = this.get('getSeriesColor');
+    var result = {};
+    this.get('allSliceLabels').forEach(function(label, iLabel) {
+      result[label] = fnGetSeriesColor(label, iLabel);
+    });
+    return result;
+  }),
+
+  fnGetSliceColor: Ember.computed('sliceColors.[]', 'getSeriesColor', function() {
+    var sliceColors = this.get('sliceColors');
+    var fnGetSeriesColor = this.get('getSeriesColor');
+
+    return function(d, i) {
+      if (d.sliceLabel && sliceColors[d.sliceLabel]) {
+        return sliceColors[d.sliceLabel];
+      } else {
+        return fnGetSeriesColor(d, i);
+      }
+    };
+  }),
+
+  // ---------------------------------------------------------------------------
+  // Legend Configuration
+  // ---------------------------------------------------------------------------
+
+  hasLegend: true,
+
+  legendItems: Ember.computed('allSliceLabels.[]', 'sliceColors', 'labelIDMapping.[]', function() {
+    var sliceColors = this.get('sliceColors');
+    return this.get('allSliceLabels').map((label) => {
+      var color = sliceColors[label];
+      return {
+        label: label,
+        fill: color,
+        stroke: color,
+        icon: () => 'square',
+        selector: ".grouping-" + this.get('labelIDMapping')[label]
+      };
+    });
+  }),
+
+  // ---------------------------------------------------------------------------
+  // Tooltip Configuration
+  // ---------------------------------------------------------------------------
+
+  showDetails: Ember.computed('isInteractive', function() {
+    if (!this.get('isInteractive')) {
+      return Ember.K;
+    }
+
+    return (data, i, element) => {
+      // Specify whether we are on an individual bar or group
+      var isGroup = Ember.isArray(data.values);
+
+      // Do hover detail style stuff here
+      element = isGroup ? element.parentNode.parentNode : element;
+      d3.select(element).classed('hovered', true);
+
+      // Show tooltip
+      var content = $('<span />');
+      if (data.barLabel) {
+        content.append($('<span class="tip-label" />').text(data.barLabel));
```

Comment:
Using jQuery to set `innerText` is actually the Karthik-approved method for avoiding reflected XSS attacks. See 1050884d5bc17658f017666fbca15605481661ad